courses/models.py

```diff
@@ -209,6 +209,17 @@ def catalog_image_url(self):
 )
 
 
+class Partner(TimestampedModel):
+    """
+    Model for course partner
+    """
+
+    name = models.CharField(max_length=255)
```

Comment:
Let's make it unique(case-insensitive), I don't think we might need to ever support partners with the same name.